cassandane/Cassandane/Instance.pm

```diff
@@ -729,6 +729,11 @@ sub _generate_imapd_conf
                 smtp_backend => 'host',
                 smtp_host => $self->{smtphost},
             );
+            if (not defined $config->get("servername")) {
+                # smtpclient needs client_bind_name or servername,
+                # falls back to servername if client_bind isn't set.
+                $config->set(servername => "127.0.0.1");
+            }
```

Comment:
And this isn't necessary because `config_servername` is always set, even when that option is blank.

This is also the wrong place to add imapd.conf defaults, because tests cannot override anything that is set here (i.e. the same problem the other PR was having with "imipnotifier: imip").  The things that are configured here are (or should be) only details that Cassandane's own infrastructure depends on, where Cassandane itself would break if a test were to configure something different.

If you need something configured for one test or one suite, add it for that test or suite.  If you need it for every (or most) tests, add it to `sub default` in `Cassandane::Config`.  But in this case it only seems necessary because the implementation in smtpclient is wrong.